datapy/Table.py

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env python3
+"""
+Table class for datapy.
+"""
+
+import os
+import sys
+import csv
+import sqlite3 as sql
+from _io import TextIOWrapper
+
+from datapy import variables # don't know if I'll need this yet
+from datapy.Schema import Schema
+
+class Table(Schema):
+    """ Table creation, deletion, and manipulation
+    Args:
+        schema_name (Schema object): the name of the schema to place the file
+        input_data (str): the path to an input file
+        table_name (str): name of table to access
+    """
+    def __init__(self, schema_name, input_data=None, table_db=None):
+        super().__init__(schema_name)
+        self.input_data = input_data
+        self.table_db = table_db
+        if table_db == None and input_data == None:
+            sys.exit("Need one of input_data or table_name")
+
+    @property
+    def table_name(self):
+        """ Determine the table_name, if a db file already exists, we use that.
+            if not, we use the name of the csv file
+        Args:
+            self.table_db (File): previously dumped sqlite database file
+            self.input_data (File): new input data in csv format
+        Returns:
+            self.table_name (str) : the name of the table
+        """
+        if self.table_db:
+            return os.path.basename(os.path.splitext(self.table_db)[0])
+        else:
+            return os.path.basename(os.path.splitext(self.input_data)[0])
+        
+    @property
+    def table_root(self):
+        """ We need all functions to be able to access the table, so we add it as a property """
```

Comment:
If using csv file for a table that didn't already exist then we should create a `.db` file